mlir/tools/mlir-rocm-runner/CMakeLists.txt

```diff
@@ -4,14 +4,48 @@ set(LLVM_OPTIONAL_SOURCES
 
 if(MLIR_ENABLE_ROCM_RUNNER)
 
+  # requirements for HIP
+  if (NOT ("AMDGPU" IN_LIST LLVM_TARGETS_TO_BUILD))
+    message(SEND_ERROR
+      "Building the mlir rocm runner requires the AMDGPU backend")
+  endif()
+
+  # Set compile-time flags for ROCm path.
+  add_definitions(-D__ROCM_PATH__="${ROCM_PATH}")
+
+  set(HIP_PATH "${ROCM_PATH}/hip" CACHE PATH " Path to which HIP has been installed")
+  set(CMAKE_MODULE_PATH "${HIP_PATH}/cmake" ${CMAKE_MODULE_PATH})
+  find_package(HIP)
+  if (NOT HIP_FOUND)
+    message(SEND_ERROR "Build the mlir rocm runner requires a working ROCm and HIP install")
+  else()
+    message(STATUS "ROCm HIP version: ${HIP_VERSION}")
+  endif()
+
+  # Locate HIP runtime library.
+  find_library(ROCM_RUNTIME_LIBRARY amdhip64
+    PATHS "${HIP_ROOT_DIR}/lib")
+  if (NOT ROCM_RUNTIME_LIBRARY)
+    message(SEND_ERROR "Could not locate ROCm HIP runtime library")
+  else()
+    message(STATUS "ROCm HIP runtime lib: ${ROCM_RUNTIME_LIBRARY}")
+  endif()
+
+  # Set HIP compile-time flags.
+  add_definitions(-D__HIP_PLATFORM_HCC__)
+  add_definitions(-D__INCLUDE_HIP__=1)
```

Comment:
... we do need to define HCC when including HIP

Reply:
We still do now? `HCC` should be a deprecated old flag and we should be using `hip-clang` now. But who knows.. if we still need it then leave it as-is.

On the other hand, get rid of `__INCLUDE_HIP__` because that's defined by us and is no longer necessary.

Reply:
We never use `hipe-clang` anywhere in the build.
So we have to specify we're targetting HCC with the flag

Reply:
Just for clarification, by default ROCM ship with `hip-clang`, not hcc, so I'm surprised we still need the "hcc" flag.

Reply:
In hip_runtime.h (rocm 5.0.1), it requires the definition of __HIP_PLATFORM_AMD/HCC__ or __HIP_PLATFORM_NVIDIA/NVCC__.  Later it does say that __HCC__ and __HIP_PLATFORM_HCC__ are deprecated.

Reply:
@zhanglx13 Sounds good, and thanks for looking into this. Please go ahead with `__HIP_PLATFORM_AMD__`.